spec/enumerable_spec.rb

```diff
@@ -0,0 +1,289 @@
+# spec/enumerable_spec.rb
+
+require '../enumerable.rb'
+
+describe Enumerable do
+  let(:num_array) { [1, 2, 3, 5, 7, 8] }
+  let(:empty_array) { [] }
+  let(:str_array) { %w[c h o c o m i n t] }
+  let(:range) { (1..10) }
+  let(:hash) { { 'a': 2, 'b': 5, 'c': 7, 'd': 8 } }
+  let(:str) { %(pineapple) }
+  let(:block_for_num) { proc { |i| i * 2 } }
+  let(:block_for_str) { proc { |item| item } }
+  let(:block_for_hash) { proc { |key, value| "#{key} is #{value}" } }
+  let(:bk_for_each_idx) { proc { |item, index| "#{index}: #{item}" } }
+  let(:bk_condit_num) { proc { |i| i.even? } }
+  let(:bk_condit_str) { proc { |item| item.is_a?(String) } }
+  let(:bk_condit_hash) { proc { |_key, value| value < 3 } }
+  let(:block_inject_num) { proc { |accum, i| accum * i } }
+  let(:block_inject_str) { proc { |accum, word| accum.size > word.size ? accum : word } }
+
+  describe '#my_each' do
+    it 'returns the same result as each method to num_array' do
+      expect(num_array.my_each(&block_for_num)).to eql(num_array.each(&block_for_num))
+    end
+
+    it 'returns the same result as each method to str_array' do
+      expect(str_array.my_each(&block_for_str)).to eql(str_array.each(&block_for_str))
+    end
+
+    it 'returns the same result as each method to range' do
+      expect(range.my_each(&block_for_num)).to eql(range.each(&block_for_num))
+    end
+
+    it 'returns the same result as each method to hash' do
+      expect(hash.my_each(&block_for_hash)).to eql(hash.each(&block_for_hash))
+    end
```

Comment:
These tests are not really checking the logic inside the block i.e., what is happening to the enumerable after `#my_each` applies the block in each iteration. You should be checking that. Right now, you are basically testing the equality of the return values of  `my_each` and `#each`, regardless of `hash` or `range` or `array`. 

Check this piece of code:

![image](https://user-images.githubusercontent.com/57726348/99217629-887cf880-27fe-11eb-804b-fe8a3bda8f8d.png)

And then check this:

![image](https://user-images.githubusercontent.com/57726348/99217721-c9750d00-27fe-11eb-89f0-ada0808f5271.png)

The above code will always print true because they both **return** the enumerable regardless of what is happening inside the block.

- [ ] Please add tests for the logic inside the block.

_Please note that this is the one place I have pointed this out. Please also add similar tests for the methods that **returns** enumerables._